openedx_tagging/core/tagging/api.py

```diff
@@ -10,33 +10,48 @@
 Please look at the models.py file for more information about the kinds of data
 are stored in this app.
 """
-from typing import List, Type
+from typing import Generator, List, Type, Union
 
 from django.db.models import QuerySet
 from django.utils.translation import gettext_lazy as _
 
-from .models import ObjectTag, Tag, Taxonomy
+from .models import ClosedObjectTag, ObjectTag, OpenObjectTag, Tag, Taxonomy
+from .registry import get_object_tag_class
 
 
 def create_taxonomy(
-    name,
-    description=None,
+    name: str,
+    description: str = None,
     enabled=True,
     required=False,
     allow_multiple=False,
     allow_free_text=False,
+    system_defined=False,
```

Comment:
@pomegranited We should not be able to create system defined in this generic api. In my PR I could create another way to create them more consistently

Reply:
Fixed with [321fd46](https://github.com/openedx/openedx-learning/pull/62/commits/321fd464bcdc4b1bc6dd39b1175180c0177b95a4).